src/linuxforhealth/csvtofhir/fhirrs/basic.py

```diff
@@ -0,0 +1,174 @@
+from importlib.resources import Resource
+from typing import Dict, List
+
+from fhir.resources.basic import Basic
+from fhir.resources.meta import Meta
+from fhir.resources.identifier import Identifier
+from fhir.resources.coding import Coding
+
+from linuxforhealth.csvtofhir.model.csv.basic import BasicCsv
+from linuxforhealth.csvtofhir.fhirutils import fhir_utils
+
+
+def convert_record(
+    group_by_key: str, record: Dict, resource_meta: Meta = None
+) -> List[Resource]:
+    resources: list = []
+
+    incoming_data: BasicCsv = BasicCsv.parse_obj(record)
+    
+    resource_id = None
+    if incoming_data.patientInternalIdentifier != None and incoming_data.patientInternalIdentifier != "":
+        resource_id = "patient-tokens." + incoming_data.patientInternalIdentifier
+    
+    # Create patient-tokens coding   
+    # "code": {
+    #    "coding": [
+    #         {
+    #             "code": "patient-tokens",
+    #             "display": "Patient Tokens",
+    #             "system": "http://ibm.com/fhir/cdm/CodeSystem/wh-basic-resource-type"
+    #         }
+    #     ],
+    #     "text": "Patient Tokens"
+    # }
+    code_cc = fhir_utils.get_codeable_concept(
+        "http://ibm.com/fhir/cdm/CodeSystem/wh-basic-resource-type",
+        "patient-tokens",
+        "Patient Tokens",
+        "Patient Tokens"
+    )
+    
+    token_cc_system = incoming_data.identifierTypeSystem
+    
+    # loop through token list and create token identifiers
+    identifierList = []
+    if incoming_data.tokenList:
+        for token in incoming_data.tokenList:
+            # tokenList list items are in the form of 
+            # ["token_name^<token_value>^system", "token_name^<token_value>^system", token_name^<token_value>^system"]
+            tokens = token.split("^")
+            token_name = tokens[0]
+            token_value = tokens[1]
+            token_system = tokens[2]
+            token_id = incoming_data.identifierNamePrefix + "." + token_name
+            
+            identifier = _build_token_identifier(token_id, token_value, token_system, token_cc_system)
+            identifierList.append(identifier)
+          
+    # create other identifier - tokenized_sid
```

Comment:
Make this code generic for any identifier by using a list similar to the token array

Reply:
Done